src/PolyType/Abstractions/IConstructorShape.cs

```diff
@@ -29,47 +29,56 @@ public interface IConstructorShape
     /// Includes all formal parameters of the underlying constructor,
     /// as well as any member that can be specified in a member initializer expression.
     /// </remarks>
-    IReadOnlyList<IParameterShape> Parameters { get; }
+    public abstract IReadOnlyList<IParameterShape> Parameters { get; }
+
+    /// <inheritdoc cref="DeclaringType"/>
+    protected abstract IObjectTypeShape DeclaringTypeNonGeneric { get; }
```

Comment:
Not sure I understand the purpose of this member?

Reply:
Is it because you can't declare a `new` virtual with the same name in the generic type?

Reply:
It's because you cannot declare an `override` *and* declare a `new virtual` with the same member name.
The way I've done it though, using `protected`, observers will have exactly the same experience as before as far as completion lists, virtual/override behavior, etc. It's only the implementations that have to override this slightly unusual name.